### PR TITLE
Migration from tensorboardX to PyTorch native Tensorboard, issue #435

### PR DESCRIPTION
**Pull request by Arquestro**
Migration from tensorboardX to PyTorch native Tensorboard.

## Description

Changes:
1. Requirements updated, minimum PyTorch version is 1.2.0 now, tensorboardX -> tensorboard.
2. All tensorboardX package references have been changed to torch.utils.tensorboard.
3. Important change in catalyst/utils/tensorboard.py: Event imported from tensorboard.compat.proto.event_b2, not from tensorboardX.proto.eventb2.

## Related Issue

https://github.com/catalyst-team/catalyst/issues/435

## Type of Change

- [X] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] I have read the [CODE_OF_CONDUCT](../CODE_OF_CONDUCT.md) document.
- [X] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [X] I have checked the code-style using `make check-style`.
- [ ] I have written the docstring in Google format for all the method and classes that I used.
- [ ] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by Arquestro** [comment]
I've ran tensorboard test from utils, seemed good to me, with no errors.
But, still afraid that Event import can break it, maybe it's only me. :)

File: requirements/requirements.txt [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-torch>=0.4.1
+torch>=1.2.0
 torchvision>=0.2.1
 imageio
 opencv-python
 scikit-image>=0.14.2
 tqdm>=4.29.1
 PyYAML
-tensorboardX>=1.8
+tensorboard
```
Comment (TezRomacH) [inline requirements/requirements.txt:10]: Does this package even exist? And is it necessary?

Comment (Arquestro) [reply requirements/requirements.txt:10]: Tensorboard package is needed for PyTorch native one i guess.
https://pytorch.org/docs/stable/tensorboard.html
If you're speaking of tensorboardX, i guess it isn't necessary to keep it.

**Review by TezRomacH — APPROVED** [review]

**Comment by Scitator** [comment]
Hi,

thinking about this PR, is it possible to save backward compatibility with some kind of try-except mechanism? or check pytorch version?
Reason: we still have a lot of users with pytorch 1.0 :)

**Comment by Arquestro** [comment]
It is possible, but i'm afraid that there will be a lot of bugs in a solution and confuse with package configurations. @TezRomacH do you have any other possible proposals on this?

**Comment by Scitator** [comment]
@Arquestro what type of package configurations confuse?
looks like you need just to check PyTorch version, if it is below 1.2.0 – used tensorboardX.
We still can have tensorboardX in the requirements for some time.

**Comment by Arquestro** [comment]
Well then none, if we can keep tensorboardX. But there will be one excess package in requirements - tensorboard or tensoboardX.

I'll try to implement and test the solution proposed anyway. :)

File: catalyst/contrib/scripts/project_embeddings.py [hunk 2]
```diff
@@ -6,7 +6,13 @@
 import numpy as np
 import pandas as pd
 import torch
-from torch.utils.tensorboard import SummaryWriter
+
+# Native tensorboard support from 1.2.0 version of PyTorch
```
Comment (TezRomacH) [inline catalyst/contrib/scripts/project_embeddings.py]: can we move it to some utils module?

and in all the other places, `from catalyst.utils.tensorboard import SummaryWriter`

**Review by TezRomacH — CHANGES_REQUESTED** [review]

File: catalyst/contrib/scripts/project_embeddings.py (hunk 2 above)
Comment (Arquestro) [reply catalyst/contrib/scripts/project_embeddings.py]: Good idea, I'll move this if construction to catalyst.utils.tensorboard.

**Review by TezRomacH — APPROVED** [review]

**Comment by Scitator** [comment]
Really cool PR, thanks!

**Pull request merged**